pangeo_forge_recipes/patterns.py

```diff
@@ -267,4 +267,10 @@ def prune_pattern(fp: FilePattern, nkeep: int = 2) -> FilePattern:
         else:  # pragma: no cover
             assert "Should never happen"
 
-    return FilePattern(fp.format_function, *new_combine_dims)
+    return FilePattern(
+        fp.format_function,
+        *new_combine_dims,
+        fsspec_open_kwargs=fp.fsspec_open_kwargs,
+        query_string_secrets=fp.query_string_secrets,
+        is_opendap=fp.is_opendap,
+    )
```

Comment:
Just noting that, _if_ `FilePattern` were a dataclass, we could just call `replace` here. Because it is not, we have to remember to update this method every time we change the signature of `FilePattern.__init__`.

Reply:
Further discussion of this in https://github.com/pangeo-forge/pangeo-forge-recipes/pull/215#discussion_r719624247 (noting for anyone who comes across this in the future)